src/workerd/util/duration-alert.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "kj/string.h"
+#include "kj/time.h"
+#include <kj/debug.h>
+
+class durationAlert {
+public:
+  durationAlert(const kj::MonotonicClock& clock, kj::Duration logWarningTime, kj::StringPtr logMessage)
+    : logWarningTime(logWarningTime),
+      logMessage(logMessage),
+      start(clock.now()),
+      clock(clock) {}
+
+  ~durationAlert() {
+    if (start - clock.now() >= logWarningTime) {
+      KJ_LOG(WARNING, "NOSENTRY", logMessage);
```

Comment:
nit: You can use `LOG_NOSENTRY` here.

Reply:
Hmm, I tried, but it doesn't seem to play nice with passing the logMessage as an argument rather than a string literal. Don't really know why that is. I'd assume b/c of it being a macro defined in terms of another macro perhaps?

Reply:
Yeah, `LOG_NOSENTRY` does its concatenation by putting two string literals next to each other to make a new string literal -- e.g. `("a" "b")` -- so it would only be expected to work when the first macro argument is itself a string literal, not a variable.

I suppose it also might be worth considering using some sort of limiting mechanism like `LOG_ERROR_PERIODICALLY` does, if we expect a delay would cause a flood of these messages at once in syslog.  But maybe it's hard to judge up front how necessary that might be.